catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc

```diff
@@ -0,0 +1,144 @@
+| *exceptionHandler* (consumer) | To let the consumer use a custom ExceptionHandler. Notice if the option bridgeErrorHandler is enabled then this option is not in use. By default the consumer will deal with exceptions, that will be logged at WARN or ERROR level and ignored. |  | ExceptionHandler
+| *exchangePattern* (consumer) | Sets the exchange pattern when the consumer creates an exchange. There are 3 enums and the value can be one of: InOnly, InOut, InOptionalOut |  | ExchangePattern
+| *block* (producer) | If sending a message to a direct endpoint which has no active consumer, then we can tell the producer to block and wait for the consumer to become active. | true | boolean
+| *failIfNoConsumers* (producer) | Whether the producer should fail by throwing an exception, when sending to a kamelet endpoint with no active consumers. | true | boolean
+| *lazyStartProducer* (producer) | Whether the producer should be started lazy (on the first message). By starting lazy you can use this to allow CamelContext and routes to startup in situations where a producer may otherwise fail during starting and cause the route to fail being started. By deferring this startup to be lazy then the startup failure can be handled during routing messages via Camel's routing error handlers. Beware that when the first message is processed then creating and starting the producer may take a little time and prolong the total processing time of the processing. | false | boolean
+| *timeout* (producer) | The timeout value to use if block is enabled. | 30000 | long
+|===
+// endpoint options: END
+
+[NOTE]
+====
+The *kamelet* endpoint is *lenient*, which means that the endpoint accepts additional parameters that are passed to the ROute Templkate engine and consumed upon route materialization.
+====
+
+== Samples
+
+
+Kamelets can be used as if they were standard Camel components.
+For example, suppose that we have created a Route Template as follows:
+
+[source,java]
+----
+routeTemplate("setBody")
+    .templateParameter("bodyValue")
+    .from("kamelet:source")
+        .setBody().constant("{{bodyValue}}");
+----
+
+[IMPORTANT]
+====
+To let the *Kamelet* component wiring the materialized route to the caller processor, we need to be able to identify the input and output endpoint of the route and this is done by using `kamele:source` to mark the input endpoint and `kamelet:sink` for the output endpoint.
+====
+
+Then the template can be instantiated and invoked as shown below:
+
+[source,java]
+----
+from("direct:template")
+    .to("kamelet:setBody?bodyValue=kamelet");
```

Comment:
Can we use mykamelet as bodyValue to make it more obvious that kamelet is not some special value